### ISSUE TITLE
[BUG] ccui.EditBox - Black Label When Editing Finished

### ISSUE THREAD

**Issue by xiaoxiangzi** [cocos2d/cocos2d-x#14976]
![7](https://cloud.githubusercontent.com/assets/5449223/12601879/be99f576-c4df-11e5-8255-6b8f2a61e229.png)
Version: cocos2d-x-3.7
Platform: android
Devices: Nexus 5, Huawei Nexus 6P
Issue: When editing finish (after enter clicked), label turns to black block. Sometimes, crashed directly.

After checking logcat, found that editBoxCallbackFunc (in file UIEditBoxImpl-android.cpp) used to refresh the text of the EditBox were invoked in different thread, neither in GL thread nor in UI thread when editing finished occasionally.

http://discuss.cocos2d-x.org/t/bug-ccui-editbox-black-label-when-editing-finished/26520

**Comment by zilongshanren** [comment]
``` java
                            mCocos2dxActivity.runOnGLThread(new Runnable() {
                                @Override
                                public void run() {
                                    Cocos2dxEditBoxHelper.__editBoxEditingDidEnd(index, editBox.getText().toString());
                                }
                            });

```

It's called in GL thread.

**Comment by xiaoxiangzi** [comment]
@zilongshanren 
Yes. In this code, it should be called at GL thread. But, actually, sometimes it was not called at GL thread, tid not the same, the label turns black:

Log below:

![image](https://cloud.githubusercontent.com/assets/5449223/12634097/a30a3758-c5b1-11e5-846f-9a1d06010b45.png)

GL thread, tid 1282
editBoxCallbackFunc called tid 1278

**Comment by zilongshanren** [comment]
@xiaoxiangzi 
I will investigate this issue later.  Try to use the latest v3.10 and test it in your device, thanks.

**Comment by zilongshanren** [comment]
@xiaoxiangzi 
Wired,  thanks.

**Comment by xiaoxiangzi** [comment]
@zilongshanren 
You're welcome.

**Comment by zilongshanren** [comment]
@xiaoxiangzi 
I have tested the latest v3.10 on Nexus 5 but it works great, no crash...
please recheck your code logic or you could send your code to me.

thanks.

**Comment by Garretthh07** [comment]
I encountered the same problem on Red Mi 2, I use Cocos2d v3.60

**Comment by Garretthh07** [comment]
I have watched cocos2d v3.10, and what's the mean of `Set the font. Only system font is allowed.`
